pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -3353,7 +3355,8 @@ public void revertReplaceSegments(String tableNameWithType, String segmentLineag
         }
 
         // Update segment lineage entry to 'REVERTED'
-        updateSegmentLineageEntryToReverted(tableNameWithType, segmentLineage, segmentLineageEntryId, lineageEntry);
+        deleteOrUpdateSegmentLineageEntryToReverted(tableNameWithType, segmentLineage, segmentLineageEntryId,
```

Comment:
Suggest keeping the `deleteOrUpdateSegmentLineageEntryToReverted` and `updateSegmentLineageEntryToReverted` as 2 separate methods for clarity

Reply:
Updated